### PR TITLE
Feature/setup scripts

### PR DESCRIPTION
**Pull request by elarib**
## Summary
A one line command installer of starlake

Tested on Windows: `Invoke-Expression (Invoke-WebRequest -Uri "https://gist.githubusercontent.com/elarib/5193d6a6f973245cc3cc95b5aac683c9/raw/aeda4718a40700a2a8bc1a9c3c6e62b4f7a5440d/setup.ps1").Content
`
Tested on linux & macosx : `sh <(curl https://gist.githubusercontent.com/elarib/5193d6a6f973245cc3cc95b5aac683c9/raw/aeda4718a40700a2a8bc1a9c3c6e62b4f7a5440d/install.sh)`

### CODE REVIEW

File: distrib/setup.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+
+add_starlake_to_path() {
+    if [ -f "$HOME/.zshrc" ] || [ -f "$HOME/.bashrc" ]; then
+        if [ -f "$HOME/.zshrc" ]; then
+            if ! grep -q "$INSTALL_DIR" ~/.zshrc; then
+                echo -e "\nexport PATH=$INSTALL_DIR:\$PATH" >> ~/.zshrc
+            fi
+            zsh ~/.zshrc
```
Comment (hayssams) [inline distrib/setup.sh:91]:
```
if [[ `echo $0` == *zsh* ]]; then  zsh ~/.zshrc; fi
```

File: distrib/setup.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        fi
+        if [ -f "$HOME/.bashrc" ]; then
+            if ! grep -q "$INSTALL_DIR" ~/.bashrc; then
+                echo -e "\nexport PATH=$INSTALL_DIR:\$PATH" >> ~/.bashrc
+            fi
+            source ~/.bashrc
```
Comment (hayssams) [inline distrib/setup.sh:97]:
```
if [[ `echo $0` == *bash* ]]; then  source ~/.bashrc; fi
```

**Review by hayssams — APPROVED** [review]
I suggest the following changes before you can merge. What do you think ?

File: distrib/setup.sh (hunk 1 above)
Comment (elarib) [reply distrib/setup.sh:91]: Interesting, i didn't know it exists.
I test it in the terminal it works, but wehn i a dd it in the bash it does not work. I ended up using 
`if [[ "$SHELL" == *zsh* ]], ...`
 WDYT ?

File: docs/docs/quickstart/010.install.mdx [hunk 3]
```diff
@@ -11,42 +11,18 @@ The script will in turn download required dependencies and copy them to the bin
...
-    0.8.1
-    0.8.0
-    0.7.4
-    0.7.3
-    Which version do you want to install? [1.0.0]:
-
-
+```shell
+sh <(curl https://raw.githubusercontent.com/starlake-ai/starlake/master/distrib/setup.sh)
+starlake help
```
Comment (elarib) [inline docs/docs/quickstart/010.install.mdx:16]: @hayssams i removed the example, should we leave it ?

Comment (hayssams) [reply docs/docs/quickstart/010.install.mdx:16]: @elarib yes!
Better leave it

**Comment by tiboun** [comment]
@elarib do you know if we need more permissions to run ps1 script ? I feel like running powershell command throught bat requires less permission. If not so, maybe we should translate the starlake.bat into ps1.

**Comment by elarib** [comment]
> @elarib do you know if we need more permissions to run ps1 script ? I feel like running powershell command throught bat requires less permission. If not so, maybe we should translate the starlake.bat into ps1.

Really, no idea. You're talking about starlake.cmd not the .bat ?

File: docs/docs/quickstart/010.install.mdx (hunk 3 above)
Comment (hayssams) [reply docs/docs/quickstart/010.install.mdx:16]: I mean keep the changes you made

**Comment by tiboun** [comment]
Yes, sorry

**Comment by hayssams** [comment]
I see that no blocking request has been made.
If you don't mind I am merging this

**Pull request merged**